ui/page/root/wallet_settings_page.go

```diff
@@ -191,101 +187,167 @@ func (pg *WalletSettingsPage) layoutMobile(gtx C, body layout.Widget) D {
-						}
-
-						peerAddrRow := clickableRowData{
-							title:     values.String(values.StrPeer),
-							clickable: pg.updateConnectToPeer,
-							labelText: pg.peerAddr,
-						}
-						return pg.clickableRow(gtx, peerAddrRow)
-					}),
-				)
-			}),
-		)
-	}
-
 	return func(gtx C) D {
-		return pg.pageSections(gtx, values.String(values.StrGeneral), dim)
+		return layout.Inset{Bottom: values.MarginPadding20}.Layout(gtx, func(gtx C) D {
+			return cryptomaterial.LinearLayout{
+				Orientation: layout.Vertical,
+				Width:       cryptomaterial.MatchParent,
+				Height:      cryptomaterial.WrapContent,
+				Background:  pg.Theme.Color.Surface,
+				Direction:   layout.Center,
+				Alignment:   layout.Middle,
+				Border:      cryptomaterial.Border{Radius: cryptomaterial.Radius(14)},
+				Padding:     layout.UniformInset(values.MarginPadding24),
+			}.Layout(gtx,
+				layout.Rigid(func(gtx C) D {
+					return pg.pageSections(gtx, values.String(values.StrGeneral), func(gtx C) D {
+						return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+							layout.Rigid(func(gtx C) D {
+								if pg.WL.SelectedWallet.Wallet.IsWatchingOnlyWallet() {
+									return D{}
+								}
+								return layout.Inset{}.Layout(gtx, pg.sectionContent(pg.changePass, values.String(values.StrSpendingPassword)))
+							}),
+							layout.Rigid(pg.sectionContent(pg.changeWalletName, values.String(values.StrRenameWalletSheetTitle))),
+							layout.Rigid(func(gtx C) D {
+								if pg.wallet.GetAssetType() == libutils.DCRWalletAsset {
+									return pg.subSection(gtx, values.String(values.StrUnconfirmedFunds), pg.spendUnconfirmed.Layout)
+								}
+								return D{}
+							}),
+							layout.Rigid(func(gtx C) D {
+								if pg.wallet.GetAssetType() == libutils.DCRWalletAsset {
+									return pg.subSection(gtx, values.String(values.StrAllowSpendingFromUnmixedAccount), pg.spendUnmixedFunds.Layout)
+								}
+								return D{}
+							}),
+							layout.Rigid(func(gtx C) D {
+								return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+									layout.Rigid(pg.subSectionSwitch(values.String(values.StrConnectToSpecificPeer), pg.connectToPeer)),
+									layout.Rigid(func(gtx C) D {
+										if pg.WL.SelectedWallet.Wallet.ReadStringConfigValueForKey(sharedW.SpvPersistentPeerAddressesConfigKey, "") == "" && pg.isPrivacyModeOn() {
+											return D{}
+										}
+
+										peerAddrRow := clickableRowData{
+											title:     values.String(values.StrPeer),
+											clickable: pg.updateConnectToPeer,
+											labelText: pg.peerAddr,
+										}
+										return pg.clickableRow(gtx, peerAddrRow)
+									}),
+								)
+							}),
+						)
+					})
+				}),
+			)
+		})
 	}
 }
 
 func (pg *WalletSettingsPage) account() layout.Widget {
-	dim := func(gtx C) D {
-		return pg.accountsList.Layout(gtx, len(pg.accounts), func(gtx C, a int) D {
-			return pg.subSection(gtx, pg.accounts[a].Name, pg.Theme.Icons.ChevronRight.Layout24dp)
-		})
-	}
```

Comment:
Keep this please (in all the methods where `dims` was defined). Makes your changes easier to review. Also makes the code easier to read when there is fewer nesting.